rspace/src/main/scala/coop/rchain/rspace/nextgenrspace/history/PointerBlock.scala

```diff
@@ -0,0 +1,42 @@
+package coop.rchain.rspace.nextgenrspace.history
+import coop.rchain.rspace.Blake2b256Hash
+import coop.rchain.rspace.nextgenrspace.history.History.codecTrie
+import scodec.Codec
+import scodec.bits.BitVector
+import scodec.codecs.{provide, vectorOfN}
+import coop.rchain.shared.AttemptOps._
+
+final case class PointerBlock private (toVector: Vector[Trie]) {
+  def updated(tuples: List[(Int, Trie)]): PointerBlock =
+    new PointerBlock(tuples.foldLeft(toVector) { (vec, curr) =>
+      vec.updated(curr._1, curr._2)
+    })
+
+  def countNonEmpty: Int = toVector.count(_ != EmptyTrie)
+}
+
+object PointerBlock {
+  val length = 256
+
+  def create(): PointerBlock = new PointerBlock(Vector.fill(length)(EmptyTrie))
```

Comment:
Now, contrary to the logical structure - which I very much like - we should probably make sure that EmptyTrie-s are not serialized and that we indeed benefit from the PointerBlock's sparsity.

More importantly though, we should make sure that the EmptyTrie case object is not being instantiated by reflection during deserialization, as this was leading to veeeeery strange hashCode/equals behavior... 

We've seen it in @lukasz-golebiewski's kryo PRs and sth similar seems to be described in https://blog.softwaremill.com/is-your-scala-object-always-a-singleton-cb3fd24a2fd9

Reply:
So I guess making the serialized structure sparse (only including NonEmptyTrie-s) would kill two birds with one stone

Reply:
https://rchain.atlassian.net/browse/RCHAIN-3226

Reply:
Some dead cell in my mind assumed that Vector is sparse. There drivers for making PointerBlock use a sparse representation that I've mentioned throught the review sitll hold though. We could probably even get rid of PointerBlock and just make Node have a sparse structure of trie pointers